Improve guide for TLS configuration of the appliance (#219)
canonical/anbox-cloud-docs: howto/anbox/tls-for-appliance.md

Setting up DNS redirection depends on your DNS provider. Refer to the documentation of your provider to create a DNS record pointing to the IP/DNS of the AWS instance where the Anbox Cloud Appliance is running.

(ref-appliance-tls-location)=
### Configure the location

> Configure the location for the appliance using the created DNS name by running the following command

Reviewer: ```suggestion
Configure the location for the appliance using the created DNS name by running the following command:
```

   sudo anbox-cloud-appliance config set network.location=your.dns.name

> The change will be automatically applied and causes all services components of the appliance to restart. If you want to defer the restart to a later point, you can use the `--no-restart` option.

Reviewer: ```suggestion
The change will be automatically applied and will cause all services components of the appliance to restart. If you want to defer the restart to a later point, you can use the `--no-restart` option.
```

    sudo cp /etc/letsencrypt/live/<your domain name>/fullchain.pem /var/snap/anbox-cloud-appliance/common/daemon/server.crt
    sudo cp /etc/letsencrypt/live/<your domain name>/privkey.pem /var/snap/anbox-cloud-appliance/common/daemon/server.key

Then restart the appliance service to make it load the new key and certificate:

    sudo snap restart anbox-cloud-appliance.daemon

With the certificate installed on the appliance, you now can access the appliance using the created domain name.

### Renew the SSL certificate

> The `certbot` snap packages installed on your machine already set up a systemd timer that will automatically renew your certificates before they expire. However, in order to get the certificate renewed successfully for the appliance, you can create `post-hook` hook for certbot which will automatically reconfigure it:

Reviewer: Did you mean a post-start hook?

```suggestion
The `certbot` snap packages installed on your machine would have already set up a systemd timer that automatically renews your certificates before they expire. However, to get the certificate renewed successfully for the appliance, you can create `post-start` hook for `certbot` which will automatically reconfigure it:
```

   ```bash
   cat <<EOF | sudo tee /etc/letsencrypt/renewal-hooks/post/001-start-appliance.sh
   #!/bin/bash
   sudo cp /etc/letsencrypt/live/<your domain name>/fullchain.pem /var/snap/anbox-cloud-appliance/common/daemon/server.crt
   sudo cp /etc/letsencrypt/live/<your domain name>/privkey.pem /var/snap/anbox-cloud-appliance/common/daemon/server.key
   sudo snap restart anbox-cloud-appliance.daemon
   EOF
   sudo chmod +x /etc/letsencrypt/renewal-hooks/post/001-start-appliance.sh
   ```

```{note}

> The appliance will be restarted at the end of the renewal of the SSL certificate to let the reverse proxy reload the certificate.

Reviewer: ```suggestion
The appliance will be restarted when the renewal of the SSL certificate is complete, to let the reverse proxy reload the certificate.
```
